exporter/signalfxexporter/factory_test.go

```diff
@@ -134,10 +137,10 @@ func TestFactory_CreateMetricsExporterFails(t *testing.T) {
 		{
 			name: "negative_duration",
 			config: &Config{
-				ExporterSettings: config.NewExporterSettings(config.NewComponentID(typeStr)),
-				AccessToken:      "testToken",
-				Realm:            "lab",
-				TimeoutSettings:  exporterhelper.TimeoutSettings{Timeout: -2 * time.Second},
+				ExporterSettings:   config.NewExporterSettings(config.NewComponentID(typeStr)),
+				AccessToken:        "testToken",
+				Realm:              "lab",
+				HTTPClientSettings: confighttp.HTTPClientSettings{Timeout: -2 * time.Second},
 			},
 			errorMessage: "failed to process \"signalfx\" config: cannot have a negative \"timeout\"",
```

Comment:
Can you also include a test for zero duration?

Reply:
This unit test is for testing failure, but as per current implementation for zero value we will not get an error and therefore we think we should not add it in this test. As the scope of this PR was to refactor the code to use HTTPConfigSettings, we haven’t changed the main implementation logic of this.
@dmitryax Please share your thoughts.

Reply:
A zero value is an interesting case since if directly setting the raw http client mean that the server must immediately return a result (which is physically possible) so it should be considered an error :)